### ISSUE TITLE
peernumber and peerindex get mixed up for conferences

### ISSUE THREAD

**Issue by zoff99** [TokTok/c-toxcore#1438]
here we give back peerindex (index):
https://github.com/TokTok/c-toxcore/blob/7c0eb5c70bea026eca665466b05213e7b21e3cba/toxcore/group.c#L2754

but here we say its peer_number:
https://github.com/TokTok/c-toxcore/blob/7c0eb5c70bea026eca665466b05213e7b21e3cba/toxcore/tox.c#L270

adding a peer add the peer to the end of the peerlist, but peernumber can a value different from the location number (peerindex) in the membuffer:
https://github.com/TokTok/c-toxcore/blob/7c0eb5c70bea026eca665466b05213e7b21e3cba/toxcore/group.c#L648

removing a peer shows it best:

if you have this group:
[0|1|2|3|4|5]
then you ```delpeer``` peer with peerindex 2, you get:
[0|1|5|2|3|4] this list

**Comment by zoff99** [comment]
here the clients can only get the number of peers in a group:

https://github.com/TokTok/c-toxcore/blob/7c0eb5c70bea026eca665466b05213e7b21e3cba/toxcore/tox.h#L2604-L2610

but are the peernumbers really the same as their peerindex?
```integers less than this number are the valid values of peer_number``` --> is this really true?

**Comment by zoff99** [comment]
new peers get a random number as peernumber:

https://github.com/TokTok/c-toxcore/blob/7c0eb5c70bea026eca665466b05213e7b21e3cba/toxcore/group.c#L1988-L2001

this may be handled better. similar as we do it with friendnumbers.

**Comment by zoff99** [comment]
@iphydf if we are using peerindex instead of peernumber, groupAV will be hard to make independent of toxcore internals.

**Comment by zugz** [comment]
Yes, the peer_numbers in group.c are not exposed in the external API, 
and it's just an (unfortunate) coincidence of names that the API uses 
peer_number to mean something else.

**Comment by zoff99** [comment]
sadly it means that every time a peer leaves the group lots of "peer_numbers" change, which for audio group incoming audio is not optimal

**Comment by zugz** [comment]
>sadly it means that every time a peer leaves the group lots of 
>"peer_numbers" change,

That's true. But you can use the pubkey to keep track of peers.
